httpbinding/user_agent_test.go

```diff
@@ -0,0 +1,12 @@
+package httpbinding
+
+import "testing"
+
+func TestUserAgentBuilder(t *testing.T) {
+	b := NewUserAgentBuilder()
+	b.AddKeyValue("foo", "1.2.3")
+	b.AddKey("baz")
+	if e, a := "foo/1.2.3 baz", b.Build(); e != a {
+		t.Errorf("expect %v, got %v", e, a)
+	}
```

Comment:
Nit: Are multiple entries possible? If so, should we add a simple test for them?

Reply:
AddKey and AddKeyValue are the only two methods, and this is a sufficient test to cover them. There is no logic that limits the number that can be added.